program/src/main/ExamImage.vala

```diff
@@ -367,111 +337,99 @@ public class ExamImage
                     }
                     break;
                 }
+
+                case Gdk.Key.n: 
+                { 
+                    if (this.is_setting_up_bounds) { 
+                        // Prevents user from adding a new question if not all questions have updated bounds. +1 since examQuestionSet includes the name
+
+                        if(System.examQuestionSet.size == System.question_incrementer+1) { 

```

Comment:
Now that they are in different files, instead of directly accessing values in System, we should make those values private and access them through getters. You don't need to change all of them, only the ones you're using.

Reply:
I started doing this but I think it might be a better idea to do this all in one go as a new PR on the decoupling branch after this gets merged in, just so we can separate those concerns and better organize the code. I'll get on that ASAP after this branch gets merged in before I forget where everything is called. This is a great idea tho 👍